plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -331,27 +517,58 @@ func (p *PowerStat) addCPUC6StateResidencyMetric(cpuID string, acc telegraf.Accu
 	acc.AddGauge("powerstat_core", fields, tags)
 }
 
-func (p *PowerStat) addCPUBusyCyclesMetric(cpuID string, acc telegraf.Accumulator) {
+func (p *PowerStat) addCPUC0StateResidencyMetric(cpuID string, acc telegraf.Accumulator) {
 	coresData := p.msr.getCPUCoresData()
 	// Avoid division by 0
 	if coresData[cpuID].timeStampCounterDelta == 0 {
 		p.Log.Errorf("timestamp counter on offset %s should not equal 0 on cpuID %s",
 			timestampCounterLocation, cpuID)
 		return
 	}
-	busyCyclesValue := roundFloatToNearestTwoDecimalPlaces(percentageMultiplier *
+	c0Value := roundFloatToNearestTwoDecimalPlaces(percentageMultiplier *
 		float64(coresData[cpuID].mperfDelta) / float64(coresData[cpuID].timeStampCounterDelta))
 	cpu := p.cpuInfo[cpuID]
 	tags := map[string]string{
 		"package_id": cpu.physicalID,
 		"core_id":    cpu.coreID,
 		"cpu_id":     cpu.cpuID,
 	}
-	fields := map[string]interface{}{
-		"cpu_busy_cycles_percent": busyCyclesValue,
+	if p.cpuC0StateResidency {
+		fields := map[string]interface{}{
+			"cpu_c0_state_residency_percent": c0Value,
+		}
+		acc.AddGauge("powerstat_core", fields, tags)
+	}
+	if p.cpuBusyCycles {
+		deprecatedFields := map[string]interface{}{
+			"cpu_busy_cycles_percent": c0Value,
+		}
+		acc.AddGauge("powerstat_core", deprecatedFields, tags)
 	}
+}
 
-	acc.AddGauge("powerstat_core", fields, tags)
+func (p *PowerStat) parsePackageMetricsConfig() {
+	if p.PackageMetrics == nil {
+		return
+	}
+	// if Package Metric config is empty, use the default settings. If not, no metric name provide means not displaying it
+	p.packageCurrentPowerConsumption = false
+	p.packageCurrentDramPowerConsumption = false
+	p.packageThermalDesignPower = false
```

Comment:
The default value is used in case the variable `p.PackageMetrics` is nil (that is if we do not set `package_metrics` in the configuration file). If the value of `p.PackageMetrics` is not nil (it means that we set `package_metrics` in the configuration file), we want only those metrics that were set by the user to be collected (so before reading which metrics are to be collected, we must set all of them to false).

Reply:
> The default value is used in case the variable p.PackageMetrics is nil (that is if we do not set package_metrics in the configuration file).

This is not correct. In your sample config you have:

```
  # package_metrics = ["current_power_consumption", "current_dram_power_consumption", "thermal_design_power"]
```

What is defined in the sample config, is the default value for something. That means that anytime someone uses the `intel_powerstat` plugin, without defining the `package_metrics` field, it should be initialized with these three values. 

Said another way, `PackageMetrics` should never be nil. It will always be a slice, either with those three strings already in it, or with whatever the user-defined it as. I believe you have updated the README to align with this, but the code does not show that.

Reply:
I changed the way of setting default values. Does it look better now?